src/scrapers/interface.ts

```diff
@@ -137,6 +137,7 @@ export interface ScraperScrapingResult {
 
 export interface Scraper<TCredentials extends ScraperCredentials> {
   scrape(credentials: TCredentials): Promise<ScraperScrapingResult>;
+  onProgress(func: (companyId: CompanyTypes, payload: {type: ScraperProgressTypes}) => void): void;
```

Comment:
done 👍

Reply:
Can you also add a test similar to what I did in my closed PR?

Reply:
To be honest, I think using typescript is enough in this case - your test would have passed before and after my change, if the test was written in javascript - it provided value for you because it probably didn't compile.
The real problem is that "onProgress" is a undocumented and untested public API, and we should write a single real test for it (In a future PR 😉 )

Reply:
I agree that tests are not the best place to test types, but having a `.ts` test for the public API that will explicitly have the usage assumptions will make a shorter dev loop if it fails in `israeli-bank-scrapers` and not in dependent packages.

> The real problem is that "onProgress" is a undocumented and untested public API, and we should write a single real test for it (In a future PR 😉 )

I agree

Reply:
@orzarchi thanks
@esakal feel free to merge